src/dialog.js

```diff
@@ -435,58 +439,81 @@ genotet.dialog.upload_ = function() {
+            modal.find('.progress').children('.progress-bar')
+              .css('width', widthPercent);
+          }
         }).done(function(data) {
             if (!data.success) {
+              modal.modal('hide');
               genotet.error('failed to upload data', data.message);
             } else {
               genotet.success('data uploaded');
             }
           })
           .fail(function(res) {
+            modal.modal('hide');
             genotet.error('failed to upload data');
           });
-        genotet.dialog.uploadProgress_(fileName);
+        if (fileType == 'bed' || fileType == 'binding') {
+          genotet.dialog.processProgress_(fileName, uploadPercent);
+        }
       });
     });
 };
 
 /**
- * Creates a dialog for uploading progress.
- * @param {string} fileName File name of the upload file.
+ * Monitors process progress.
+ * @param {string} fileName File name of the processing file.
+ * @param {number} startNum Start percent of progress.
  * @private
  */
-genotet.dialog.uploadProgress_ = function(fileName) {
+genotet.dialog.processProgress_ = function(fileName, startNum) {
   var modal = $('#dialog');
-  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.progress,
-    function() {
-      modal.modal({
-        backdrop: 'static',
-        keyboard: false
-      });
-      var number = 0;
-      modal.find('#btn-ok').prop('disabled', true);
-      var interval = setInterval(function() {
-        number++;
-        // This is a fake progress bar.
-        // Increase 3% for the progress bar every 3 seconds.
-        // When reaching 99, do not increase it any more.
-        var widthPercent = Math.min(number * 3, 99) + '%';
-        var params = {
-          type: 'check-finish',
-          fileName: fileName
-        };
-        $.get(genotet.data.serverURL, params, function(data) {
-          if (data) {
-            clearInterval(interval);
-            widthPercent = '100%';
-            modal.find('#btn-ok').prop('disabled', false);
-            modal.find('.progress').children('.progress-bar')
-              .css('width', widthPercent);
-          }
-        });
-        modal.find('.progress').children('.progress-bar')
-          .css('width', widthPercent);
-      }, genotet.dialog.queryInterval_);
+  var number = startNum;
+  var interval = setInterval(function() {
+    number++;
+    // This is a fake progress bar.
+    // Increase 1% for the progress bar every 3 seconds.
+    // When reaching 99, do not increase it any more.
+    var widthPercent = Math.min(number, 99) + '%';
+    var params = {
+      type: 'check-finish',
+      fileName: fileName
+    };
+    $.get(genotet.data.serverURL, params, function(data) {
+      if (data) {
+        clearInterval(interval);
+        modal.modal('hide');
+      }
     });
+    modal.find('.progress').children('.progress-bar')
+      .css('width', widthPercent);
+  }, genotet.dialog.queryInterval_);
```

Comment:
Will 3*1000 be too long for the interval? It takes 3s * 30percent = 90 secs. As far as I remember processing doesn't take that long?